File: gof_elixir/lib/game_of_life/board.ex
defmodule GameOfLife.Board do
  @doc """
  Parses the board into the world.
  """
  def parse(board, world) do
    lines = String.split(board)
    values = Enum.map(lines, fn line -> String.split(line, ",") end)
    board_size = length(values)

    for x <- 0..(board_size - 1),
        y <- 0..(board_size - 1) do
          {:ok, row} = Enum.fetch(values, y)
          {:ok, string_value} = Enum.fetch(row, x)
          {value, _} = Integer.parse(string_value)
          GameOfLife.World.set(world, x, y, value)
        end
  end

  @doc """
  Generates a string that represents the world state.
  """
  def generate(world) do
    world_state = GameOfLife.World.state(world)
    world_size = map_size(world_state)

    rows = for y <- 0..(world_size - 1) do
      0..(world_size - 1)
      |> Enum.map(fn x -> GameOfLife.Cell.state(world_state, x, y) end)
      |> Enum.join(",")
    end

    Enum.join(rows, "\n")
  end
end

File: gol_ruby/lib/board.rb
require "cell"

class Board
  attr_reader :size

  def initialize(size)
    @size = size
    @board = build_board
  end

  def get(x, y)
    @board[y][x]
  end

  def to_s
    @board.map { |row| row.join(",") }.join("\n")
  end

  private

  def build_board
    Array.new(@size) do
      Array.new(@size) { Cell.new }
    end
  end
end
